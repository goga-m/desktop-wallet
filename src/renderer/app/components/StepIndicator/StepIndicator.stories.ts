import { number, withKnobs } from "@storybook/addon-knobs";

import StepIndicator from "./StepIndicator.vue";

export default { title: "Basic / StepIndicator", decorators: [withKnobs] };

export const Default = () => ({
	components: { StepIndicator },
    props: {
		activeIndex: {
			default: number("Active Index", 1),
		},
		size: {
			default: number("Size", 2),
		},
	},
	template: `
		<div class="space-x-4 p-5">
			<StepIndicator :activeIndex="activeIndex" :size="size"></StepIndicator>
		</div>
	`,
});
