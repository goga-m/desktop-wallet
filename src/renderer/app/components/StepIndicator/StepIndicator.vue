<template>
	<div class="StepIndicator flex w-full" data-testid="StepIndicator">
		<div
			v-for="index in size"
			class="StepIndicator__step flex-1 mr-2 rounded-lg"
			v-bind:key="'step' + index"
			data-testid="StepIndicator__step"
			v-bind:class="{'active': activeIndex == index}"
		></div>
	</div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from "vue-property-decorator";

@Component
export default class StepIndicator extends Vue {
	@Prop({ default: 0 }) public activeIndex!: number;
	@Prop({ default: 2 }) public size!: number;
}
</script>

<style lang="postcss" scoped>
transition div {
	transition: background-color 1s;
}

.StepIndicator__step {
	min-height: 2px;
	min-width: auto;
	background-color: theme("colors.gray.200");
	&.active {
		background-color: theme("colors.yellow.600");
	}
}
</style>
