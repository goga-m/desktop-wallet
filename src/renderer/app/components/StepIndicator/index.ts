import StepIndicator from "./StepIndicator.vue";

export { StepIndicator };
