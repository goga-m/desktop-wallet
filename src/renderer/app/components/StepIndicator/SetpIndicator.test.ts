import { shallowMount } from "@vue/test-utils";

import StepIndicator from "@/app/components/StepIndicator/StepIndicator.vue";

describe("StepIndicator", () => {
	it("should render", () => {
		const wrapper = shallowMount(StepIndicator);

		expect(wrapper.find(".StepIndicator")).toBeTruthy();
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("should render with the configured size", () => {
		const wrapper = shallowMount(StepIndicator, {
			propsData: {
				size: 3,
			},
		});

		const steps = wrapper.findAll(`[data-testid="StepIndicator__step"]`);
		expect(steps.length).toEqual(3);
	});
	
    it("should have the right active index", () => {
		const wrapper = shallowMount(StepIndicator, {
			propsData: {
				size: 3,
                activeIndex: 1,
			},
		});

		const steps = wrapper.findAll(`[data-testid="StepIndicator__step"]`);
    
		expect(steps.wrappers[0].classes()).toContain("active");
	});
});
