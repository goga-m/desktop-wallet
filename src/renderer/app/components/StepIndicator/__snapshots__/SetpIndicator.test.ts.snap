// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepIndicator should render 1`] = `
<div data-testid="StepIndicator" class="StepIndicator flex w-full">
  <div data-testid="StepIndicator__step" class="StepIndicator__step flex-1 mr-2 rounded-lg"></div>
  <div data-testid="StepIndicator__step" class="StepIndicator__step flex-1 mr-2 rounded-lg"></div>
</div>
`;
